client/signup/config/flows.js

```diff
@@ -225,6 +225,15 @@ const flows = {
 		description: 'Used by `get.blog` users that connect their site to WordPress.com',
 		lastModified: '2016-11-14'
 	},
+
+	wpcc: {
+		steps: [ 'oauth2-user' ],
+		destination: function( dependencies ) {
+			return dependencies.oauth2_client_id;
```

Comment:
This doesn't look like a valid route to me.
I think we need to keep the `oauth2_redirect` parameter to redirect the user once he signs up. We could actually only have this parameter as it can be used to deduce the client_id (as you know from your server patch).